androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/SharedPreferencesTokenStore.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.androidbrowserhelper.trusted;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.util.Base64;
+
+import androidx.annotation.Nullable;
+import androidx.browser.trusted.Token;
+import androidx.browser.trusted.TokenStore;
+
+/**
+ * Implements a {@link TokenStore} that uses {@link SharedPreferences} as the storage mechanistm
+ * for the {@link Token}.
+ */
+public class SharedPreferencesTokenStore implements TokenStore {
+    private static final String SHARED_PREFERENCES_NAME = "com.google.androidbrowserhelper";
+    private static final String KEY_TOKEN =
+            "com.google.androidbrowserhelper.trusted.SharedPreferencesTokenStore.TOKEN";
+
+    private Context mContext;
+
+    /**
+     * Creates a new SharedPreferencesTokenStore
+     *
+     * @param mContext The {@link Context} where the {@link SharedPreferences} will be stored.
+     */
+    public SharedPreferencesTokenStore(Context mContext) {
+        this.mContext = mContext;
+    }
+
+    /**
+     * This persists the given {@link Token} on a {@link SharedPreferences}.
+     * Subsequent calls will overwrite the previously given {@link Token}.
+     *
+     * @param token The token to persist. It may be {@code null} to clear the storage.
+     */
+    @Override
+    public void store(@Nullable Token token) {
+        SharedPreferences preferences =
+                mContext.getSharedPreferences(SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE);
+
+        // Clear the preference if the token is null
+        if (token == null) {
+            preferences.edit().remove(KEY_TOKEN).apply();
+            return;
+        }
+
+        String encodedToken = Base64.encodeToString(token.serialize(), Base64.DEFAULT);
```

Comment:
In Chrome [we use](https://source.chromium.org/chromium/chromium/src/+/master:chrome/android/java/src/org/chromium/chrome/browser/browserservices/permissiondelegation/TrustedWebActivityPermissionStore.java;l=244?q=trustedwebactivitypermissionstore&ss=chromium&originalUrl=https:%2F%2Fcs.chromium.org%2Fsearch%2F) `NO_WRAP | NO_PADDING`. Do you think it matters?

Reply:
Those seem to be used to save some bytes when encoding the content. It probably has larger impact when encoding things at scale. Adding this has no downsides other than making the string less human readable, which is not a requirement in this case. I'll add it.

Reply:
done